### PR TITLE
docs: Review data types

### PR DESCRIPTION
**Pull request by bjchambers**

### CODE REVIEW

File: python/docs/source/guide/data_types.md [hunk 1]
```diff
@@ -73,6 +73,16 @@ For example, `{name: string, age: u32 }` is a record type with two fields and `{
 
 NOTE: Record types may be nested.
 
+## Collection Types
+
+Kaskada also supports collections -- lists and maps.
```
Comment (jordanrfrazier) [inline python/docs/source/guide/data_types.md:78]: Well, in some cases. https://github.com/kaskada-ai/kaskada/issues/494

File: python/docs/source/examples/time_centric.ipynb [hunk 2]
```diff
@@ -173,12 +173,8 @@
...
    "metadata": {},
    "source": [
-    "Kaskada query language is parsed by the `fenl` extension. Query calculations are\n",
-    "defined in a code blocks starting with `%%fenl`.\n",
-    "\n",
-    "See [the `fenl`\n",
-    "documentation](https://kaskada-ai.github.io/docs-site/kaskada/main/fenl/fenl-quick-start.html)\n",
-    "for more information.\n",
+    "Kaskada queries are defined in Python, using the `{py}Timestream` class.\n",
+    "Sources are Timestreams generally containing [records](../guide/data_types.md#record-types).\n",
```
Comment (jordanrfrazier) [inline python/docs/source/examples/time_centric.ipynb:177]: Is there a scenario where we wouldn't interpret a `Source` input as a `record`?

**Review by jordanrfrazier — APPROVED** [review]

File: python/docs/source/examples/time_centric.ipynb (hunk 2 above)
Comment (bjchambers) [reply python/docs/source/examples/time_centric.ipynb:177]: Conceivably. If we eventually added a timeseries / sensor input, I could see a case where we had something like a `timestream of f64` (sensor readings).

File: python/docs/source/guide/data_types.md (hunk 1 above)
Comment (bjchambers) [reply python/docs/source/guide/data_types.md:78]: That's OK for the purposes of the docs.

**Pull request merged**